var/spack/repos/builtin/packages/quantum-espresso/package.py

```diff
@@ -244,9 +254,28 @@ class QuantumEspresso(Package):
           sha256='b1aa3179ee1c069964fb9c21f3b832aebeae54947ce8d3cc1a74e7b154c3c10f',
           when='+patch@6.4.1:6.5.0')
 
+    # QE 6.4.1 Fix intent for Fujitsu compiler
+    patch('fj-intent.6.4.1.patch', when='+patch@6.4.1')
+    # QE 6.4.1 Fix intent
+    patch('https://gitlab.com/QEF/q-e/-/commit/c2a86201ed72693ffa50cc99b22f5d3365ae2c2b.diff',
+          sha256='b2dadc0bc008a3ad4b74ae85cc380dd2b63f2ae43a634e6f9d8db8077efcea6c',
+          when='+patch@6.4.1')
+
+    # QE 6.4.1 Small fixes for XLF compilation
+    patch('https://gitlab.com/QEF/q-e/-/commit/cf088926d68792cbaea48960c222e336a3965df6.diff',
+          sha256='bbceba1fb08d01d548d4393bbcaeae966def13f75884268a0f84448457b8eaa3',
+          when='+patch@6.4.1:6.5.0')
+
     # Configure updated to work with NVIDIA compilers
     patch('nvhpc.patch', when='@6.5 %nvhpc')
 
+    # Configure updated to work with Fujitsu compilers
+    patch('fj.6.5.patch', when='@6.5 %fj')
+    patch('fj.6.6.patch', when='@6.6:6.7 %fj')
+
+    # extlibs_makefile updated to work with fujitsu compilers
+    patch('fj-fox.patch', when='%fj')
+
```

Comment:
The `when=` clauses here need a `+patch`

Reply:
I should add some context here. There are some 3rd party patches in QE. The one I added is for QMCPACK. We should really just upstream this code but it never happened for various reasons. The `+patch` which is the default variant is just a way to detect/label that there is a patch on the official release version of QE. So if you apply a patch from upstream or elsewhere, you should include `when='+patch'` to activate the patch.

So, `when='~patch'`, means plain vanilla QE with no patches.

Its not a great system at all, but there is no way to suppress patch on a per patch level in Spack.

Reply:
fixed.